cmd/ekco/cli/operator.go

```diff
@@ -118,3 +133,112 @@ func OperatorCmd(v *viper.Viper) *cobra.Command {
 
 	return cmd
 }
+
+type patchUInt32Value struct {
+	Op    string `json:"op"`
+	Path  string `json:"path"`
+	Value uint32 `json:"value"`
+}
+
+func startPrometheusAutoscaler(kubeClient kubernetes.Interface, restClientConfig *restclient.Config, logger *zap.SugaredLogger) error {
+	dynamicClient, err := dynamic.NewForConfig(restClientConfig)
+	if err != nil {
+		return err
+	}
+
+	installerList, err := dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "cluster.kurl.sh",
+		Version:  "v1beta1",
+		Resource: "installers",
+	}).
+		List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	if len(installerList.Items) == 0 {
+		return nil
+	}
+
+	var currentInstaller unstructured.Unstructured
+	for _, installer := range installerList.Items {
+		selectedInstallerTimestamp := currentInstaller.GetCreationTimestamp()
+		nextInstallerTimestamp := installer.GetCreationTimestamp()
+		if (&selectedInstallerTimestamp).Before(&nextInstallerTimestamp) {
+			currentInstaller = installer
+		}
+	}
+
+	spec := currentInstaller.Object["spec"].(map[string]interface{})
+	if _, ok := spec["prometheus"]; !ok {
+		logger.Info("Prometheus autoscaler disabled: Prometheus not detected in kurl install.")
```

Comment:
i think we should just look for the `monitoring` namespace and/or the prometheus k8s object instead of the kurl spec.

Reply:
I was originally looking for the monitoring namespace but I was concerned that an existing cluster could potentially have this namespace.  What would you say the advantages of using the namespace or crd has over this?  Shouldn't this logic only even been active when the cluster was deployed from kurl with both prometheus and ekco?

Reply:
you are already hardcoding the namespace [here](https://github.com/replicatedhq/ekco/pull/34/files#diff-5fb07fb0df8113f9959f467408fa6ddd815181a97ed88283feed969e5bd5467fR227). i agree with salah although i'd argue we should go further and discover the prom namespace if possible.

Reply:
no need to discover prom namespace. i just checked and its hardcoded in kurl unlike some other addons.